AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthenticationProviderAdapter+SignOut.swift

```diff
@@ -27,11 +27,19 @@ extension AuthenticationProviderAdapter {
         // There is no point on returning an error back to the developer, because
         // they do not control how the UI is presented.
         dispatchPrecondition(condition: .onQueue(DispatchQueue.main))
+
         let signOutOptions = SignOutOptions(signOutGlobally: true, invalidateTokens: true)
-        awsMobileClient.signOut(options: signOutOptions) { error in
+        awsMobileClient.signOut(options: signOutOptions) { [weak self] error in
             guard error == nil else {
                 let authError = AuthErrorHelper.toAuthError(error!)
-                completionHandler(.failure(authError))
+                if case .notAuthorized(_, _, _) = authError {
+                    // signOut globally might return notAuthorized when the current token is expired or invalidated
+                    // In this case, we just signOut the user locally and return a success result back.
```

Comment:
This is misleading if the contract is that `signOut` does a global signout. Let's document this, ensure the behavior is consistent across platforms, and emit a Hub event if we have to fall back to "local sign out only"

Reply:
Made the default behavior of signOut to local signOut and added an option to globalSignOut.
We are handling `notAuthorized` since its not a recoverable error.